crates/api/src/op_store.rs

```diff
@@ -61,6 +63,7 @@ impl PartialOrd for StoredOp {
 }
 
 /// The API that a kitsune2 host must implement to provide data persistence for kitsune2.
+#[cfg_attr(any(test, feature = "mockall"), automock)]
 pub trait OpStore: 'static + Send + Sync + std::fmt::Debug {
```

Comment:
Question! Now that I've spent a bit more time getting to grips with using the core module. I'm wondering whether we want to `automock` things that have in-memory implementations? Like the op store and transport modules do.

I think the answer should be yes if we're testing errors or something else that the memory implementations won't do for us. So I think this is valid to put here but I am wondering if the tests in this PR need it still?

Reply:
This PR does not need the op store mock. It was split off from the process incoming response PR, where the mock is used to fail at writing ops in a test.